invokeai/frontend/web/src/features/gallery/components/Boards/BoardsList/BoardsList.tsx

```diff
@@ -19,53 +22,106 @@ const overlayScrollbarsStyles: CSSProperties = {
   width: '100%',
 };
 
-type Props = {
-  isOpen: boolean;
-};
-
-const BoardsList = (props: Props) => {
-  const { isOpen } = props;
+const BoardsList = () => {
   const selectedBoardId = useAppSelector((s) => s.gallery.selectedBoardId);
   const boardSearchText = useAppSelector((s) => s.gallery.boardSearchText);
+  const allowPrivateBoards = useAppSelector((s) => s.config.allowPrivateBoards);
   const queryArgs = useAppSelector(selectListBoardsQueryArgs);
   const { data: boards } = useListAllBoardsQuery(queryArgs);
   const filteredBoards = boardSearchText
```

Comment:
Probably a good idea to put the boards arrays in a `useMemo`

```suggestion
  const { filteredPrivateBoards, filteredSharedBoards } = useMemo(() => {
    const filteredBoards = boardSearchText
      ? boards?.filter((board) => board.board_name.toLowerCase().includes(boardSearchText.toLowerCase()))
      : boards;
    const filteredPrivateBoards = filteredBoards?.filter((board) => board.is_private);
    const filteredSharedBoards = filteredBoards?.filter((board) => !board.is_private);
    return { filteredPrivateBoards, filteredSharedBoards };
  }, [boardSearchText, boards]);
```

Reply:
Also I think it's a bit nicer to always return an array. Can use the const `EMPTY_ARRAY` so it has a stable reference

```suggestion
const filteredPrivateBoards = filteredBoards?.filter((board) => board.is_private) ?? EMPTY_ARRAY;
```

Then in the component remove the truthy check